src/azure-cli/azure/cli/command_modules/network/tests/latest/test_private_endpoint_commands.py

```diff
@@ -145,6 +157,104 @@ def test_private_endpoint_connection_keyvault(self, resource_group):
 
         self.cmd('network private-endpoint-connection delete --id {kv_pe_id} -y')
 
+    @ResourceGroupPreparer(name_prefix='cli_test_hsm_pe')
+    def test_hsm_private_endpoint_connection(self, resource_group):
+        self.kwargs.update({
+            'hsm': self.create_random_name('cli-test-hsm-pe-', 24),
+            'loc': 'centraluseuap',
+            'vnet': self.create_random_name('cli-vnet-', 24),
+            'subnet': self.create_random_name('cli-subnet-', 24),
+            'pe': self.create_random_name('cli-pe-', 24),
+            'pe_connection': self.create_random_name('cli-pec-', 24),
+            'rg': resource_group
+        })
+
+        # Prepare hsm and network
+        hsm = self.cmd('keyvault create --hsm-name {hsm} -g {rg} -l {loc} '
+                       '--administrators "3707fb2f-ac10-4591-a04f-8b0d786ea37d"').get_output_in_json()
+        self.kwargs['hsm_id'] = hsm['id']
+        self.cmd('network vnet create '
+                 '-n {vnet} '
+                 '-g {rg} '
+                 '-l {loc} '
+                 '--subnet-name {subnet}',
+                 checks=self.check('length(newVNet.subnets)', 1))
+        self.cmd('network vnet subnet update '
+                 '-n {subnet} '
+                 '--vnet-name {vnet} '
+                 '-g {rg} '
+                 '--disable-private-endpoint-network-policies true',
+                 checks=self.check('privateEndpointNetworkPolicies', 'Disabled'))
+
+        # Create a private endpoint connection
+        pe = self.cmd('network private-endpoint create '
+                      '-g {rg} '
+                      '-n {pe} '
+                      '--vnet-name {vnet} '
+                      '--subnet {subnet} '
+                      '-l {loc} '
+                      '--connection-name {pe_connection} '
```

Comment:
The variable name `pe` is not accurate for `--connection-name`.

```
> az network private-endpoint create -h

--connection-name                [Required] : Name of the private link service connection.
```

Should be something like `pl`.

Reply:
It's `private_endpoint`'s `private_link_service_connection`.
https://github.com/Azure/azure-cli/blob/e0d7b53da27cd6c7242456cfd8cecc00fe5fe04a/src/azure-cli/azure/cli/command_modules/network/custom.py#L3063